From fecb65813ddf52abf310bc2227a0ac869dc897d1 Mon Sep 17 00:00:00 2001
From: Christian Marangi <ansuelsmth@gmail.com>
Date: Wed, 29 Jan 2025 14:47:41 +0100
Subject: [PATCH 1/3] airoha: ethernet: drop xsi-mac reset

In preparation for support for Ethernet and PON PCS, drop the xsi-mac
reset from airoha_eth. This reset is related to the Ethernet PCS and
should be handled in the dedicated driver.

Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>
---
 drivers/net/ethernet/airoha/airoha_eth.c | 9 ++++-----
 drivers/net/ethernet/airoha/airoha_eth.h | 2 +-
 2 files changed, 5 insertions(+), 6 deletions(-)

--- a/drivers/net/ethernet/airoha/airoha_eth.c
+++ b/drivers/net/ethernet/airoha/airoha_eth.c
@@ -2948,11 +2948,10 @@ static int airoha_probe(struct platform_
 		return err;
 	}
 
-	eth->xsi_rsts[0].id = "xsi-mac";
-	eth->xsi_rsts[1].id = "hsi0-mac";
-	eth->xsi_rsts[2].id = "hsi1-mac";
-	eth->xsi_rsts[3].id = "hsi-mac";
-	eth->xsi_rsts[4].id = "xfp-mac";
+	eth->xsi_rsts[0].id = "hsi0-mac";
+	eth->xsi_rsts[1].id = "hsi1-mac";
+	eth->xsi_rsts[2].id = "hsi-mac";
+	eth->xsi_rsts[3].id = "xfp-mac";
 	err = devm_reset_control_bulk_get_exclusive(eth->dev,
 						    ARRAY_SIZE(eth->xsi_rsts),
 						    eth->xsi_rsts);
--- a/drivers/net/ethernet/airoha/airoha_eth.h
+++ b/drivers/net/ethernet/airoha/airoha_eth.h
@@ -21,7 +21,7 @@
 #define AIROHA_MAX_NUM_IRQ_BANKS	4
 #define AIROHA_MAX_DSA_PORTS		7
 #define AIROHA_MAX_NUM_RSTS		3
-#define AIROHA_MAX_NUM_XSI_RSTS		5
+#define AIROHA_MAX_NUM_XSI_RSTS		4
 #define AIROHA_MAX_MTU			9216
 #define AIROHA_MAX_PACKET_SIZE		2048
 #define AIROHA_NUM_QOS_CHANNELS		4
